### PR TITLE
feat: SASjs FileTree type

### PR DESCRIPTION
**Pull request by saadjutt01**
## Intent

Move `FileTree` type from `sasjs/server` to `sasjs\utils`. 
Also added `ServicePackSASjs` type for `SASJS` server, as per `sasjs/server` [issue](https://github.com/sasjs/server/issues/109), to have standalone build.json containing additional info other than `members`.

## Implementation

Added type with validation util modules + specs.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] All unit tests are passing (`npm test`).
- [ ] All `sasjs-cli` unit tests are passing (`npm test`).
- [x] Reviewer is assigned.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -4124,9 +4124,9 @@
       }
     },
     "node_modules/minimist": {
-      "version": "1.2.5",
-      "resolved": "https://registry.npmjs.org/minimist/-/minimist-1.2.5.tgz",
-      "integrity": "sha512-FM9nNUYrRBAELZQT3xeZQ7fmMOBg6nWNmJKTcgsJeaLstP/UODVpGsr5OhXhhXg6f+qtJ8uiZ+PUxkDWcgIXLw==",
+      "version": "1.2.6",
```
Comment (YuryShkoda) [inline package-lock.json:4127]: I don't think that `package-lock` should be modified within this PR. Please use `npm ci` command to install dependencies.

File: src/utils/isFileTree.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import {
```
Comment (YuryShkoda) [inline src/utils/isFileTree.ts:1]: I would call this utility `fileTree`

**Review by YuryShkoda — CHANGES_REQUESTED** [review]
See comments below

File: package-lock.json (hunk 1 above)
Comment (saadjutt01) [reply package-lock.json:4127]: I explicitly ran `npm audit fix` for soft fixing of vulnerabilities. Now there are 0 vulnerabilities.
There's separate commit for that.

Comment (YuryShkoda) [reply package-lock.json:4127]: aha, got it!

File: src/utils/isFileTree.ts (hunk 2 above)
Comment (saadjutt01) [reply src/utils/isFileTree.ts]: updated

**Review by YuryShkoda — DISMISSED** [review]

**Review by YuryShkoda — APPROVED** [review]

**Pull request merged**